### PR TITLE
Iteration 2 sales analyst

### PR DESCRIPTION
**Pull request by DougWelchons**
this gets us rounded out for iteration 2. All passing tests and passing spec harness through iteration 2.

### CODE REVIEW

File: test/sales_engine_test.rb [hunk 1]
```diff
@@ -6,35 +6,93 @@
...
+
+    def test_it_can_find_all_invoices_by_merchant_id
+      sales_engine = SalesEngine.from_csv({
+                                            :items     => "./data/items.csv",
+                                            :merchants => "./data/merchants.csv",
+                                            :invoices => "./data/invoices.csv",
+                                            :customers => "./data/customers.csv",
+                                          })
+
+        assert_equal 7, sales_engine.find_all_invoices_by_merchant_id(12334112).count
```
Comment (brisag) [inline test/sales_engine_test.rb]: indendtation looks fine

**Comment by avjohnston** [comment]
Awesome work, hound is a b-word so I wouldn't worry about most of those issues. The code looks good and concise and all of the test pass.

**Comment by brisag** [comment]
Looks amazing!! We will update the csvs and add the others once we do iteration 3 :)

**Comment by avjohnston** [comment]
Good catch on those line indentations. I think its ready to merge!

**Pull request merged**